### ISSUE TITLE
[bug] moon crashes when there's a environment variable entry

### ISSUE THREAD

**Issue by bradleyayers** [moonrepo/moon#1310]
**Describe the bug**

I was seeing if file groups supported environment variables (I want to re-use environment variable groups) and found that moon crashes.

```
moon run storybook-build
Error:   × Main thread panicked.
  ├─▶ at /Users/runner/work/moon/moon/nextgen/config/src/shapes/input_path.rs:51:17
  ╰─▶ internal error: entered unreachable code
  help: set the `RUST_BACKTRACE=1` environment variable to display a backtrace.

[1]    16039 abort      moon run storybook-build
```

**Steps to reproduce**

Create a file group with a `$FOO` item, e.g.

```
fileGroups:
  frontend:
    - "src/**/*"
    - "static/**/*"
    - ".*"
    - "*.*"
    - "$FOO"
```

Run a task e.g. `moon run foo`.

Removing the `$FOO` item fixes it.

**Comment by milesj** [comment]
Seems like an oversight, looking into.

**Comment by milesj** [comment]
Got this working here: https://github.com/moonrepo/moon/pull/1316

**Comment by milesj** [comment]
Fixed in v1.21
